lib/coverage.js

```diff
@@ -269,19 +269,41 @@ internals.instrument = function (filename) {
 
     let skipStart = 0;
     let segmentSkip = false;
+
+    // In addition to maintaining the current coverage bypass state, support
+    // comments that push that state to a stack, allowing transpiled code to
+    // skip coverage without interfering with a user-defined bypass block.
+    const skipStack = [];
     tree.comments.forEach((comment) => {
 
-        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on)\$\s*$/);
+        const directive = comment.value.match(/^\s*\$lab\:coverage\:(off|on|push|pop)\$\s*$/);
         if (directive) {
-            const skip = directive[1] !== 'on';
-            if (skip !== segmentSkip) {
-                segmentSkip = skip;
-                if (skip) {
-                    skipStart = comment.range[1];
+            const command = directive[1];
+            if (command === 'push') {
+                skipStack.push(segmentSkip);
+            }
+            else {
+                let newSkipState;
+                if (command === 'pop') {
+                    if (!skipStack.length) {
+                        throw new Error('unable to pop coverage bypass stack');
```

Comment:
There are a number of other reasonable behaviors here, such as just resetting to the default (`on`), or doing nothing. I lean towards this behavior because it seems more helpful if you mis-implement machine generated stack manipulation comments, though in that case it may be prudent to also warn if the stack isn't empty by the end of the file. Thoughts?